tools/convert_posts_to_page_bundles.py

```
@@ -0,0 +1,67 @@
+        parsed_url = urlparse(img_url)
+        file_extension = os.path.splitext(parsed_url.path)[1]
+        if not file_extension:
+            file_extension = '.jpg'  # Default to .jpg if no extension is found
+        local_img_path = f'image_{i + 1}{file_extension}'
+        full_img_path = os.path.join(post_dir, local_img_path)
+        if download_image(img_url, full_img_path):
+            content = content.replace(f'![{alt_text}]({img_url})', f'![{alt_text}]({local_img_path})')
+
+    return content
+
+def convert_posts_to_page_bundles(content_dir):
```

Ruff (0.8.2)

Missing return type annotation for public function `convert_posts_to_page_bundles`

Add return type annotation: `None`

(ANN201)